core/services/relay/evm/functions/logpoller_wrapper.go

```diff
@@ -419,6 +419,17 @@ func (l *logPollerWrapper) handleRouteUpdate(ctx context.Context, activeCoordina
 			l.lggr.Errorw("LogPollerWrapper: Failed to update routes", "err", err)
 		}
 	}
+
+	// Unregister filters with the logPoller for any previous coordinators that are no longer active or proposed
+	filters := l.logPoller.GetFilters()
+	for _, filter := range filters {
+		if filter.Name == filterName(l.activeCoordinator) || filter.Name == filterName(l.proposedCoordinator) {
```

Comment:
This is too risky. What if some day we have multiple jobs running on the node? You can't affect filters that are not yours here.

It would be good to have a name prefix unique to a DON but unfortunately we don't and it will be a bit painful to add it now. The next best thing I can think of is just to remove the previous one that's not active any more (won't be 100% correct but best effort). Any other ideas?